woocommerce/woocommerce-admin
# Onboarding: Update payments task flow

Fixes (part of) #3710 

This PR adds the following:
* Refactors and redesigns the payment methods list.
* Adds in done/skip options for the payments task list.
* Adds events for skip, done, and setup of payments.
* Adds a recommended payment method.
* Fixes some previous issues with error handling and pending states.

This PR does not address the following which will be handled in a follow-up PR:
* Toggles for already set up payment methods and respective events.
* Updating payments configured outside the task list.
* Adding new offline payment methods

### Questions
* Does the copy for payment labels and descriptions look okay @pmcpinto  ?
* Some of the previous methods don't have a "Skip" option, meaning there is no way to return to the payment method options after selecting a specific payment.  Should we add "Skip" options to return to payment options for each method? /cc @jameskoster 

### Screenshots
<img width="728" alt="Screen Shot 2020-02-27 at 8 39 19 PM" src="https://user-images.githubusercontent.com/10561050/75480284-43c7dd80-59a1-11ea-9d15-b2d615aa8fbd.png">

### Detailed test instructions:

1. Delete the option `woocommerce_task_list_payments` if you need to reset progress at any point.
1. Go through each of the payment methods listed below and make sure they work as expected with both configuration options if multiple exist.
1. Make sure the recommended gateway ("Stripe") has a ribbon if not yet configured.
1. Make sure the event `tasklist_payment_done` fires with configured options when clicking "Done."
1. Make sure the event `tasklist_payment_skip_task` fires with available options when clicking "My store doesn’t take payments."
1. Make sure the event `tasklist_payment_setup` fires when clicking "Set up" on a payment with the available options and clicked option.

Payment method requirements:
* Stripe - requires store to be supported Stripe country, e.g. US. Test manual and oauth option by connecting Jetpack and activating WCS.
* Paypal - test oauth and manual connection.  Manual can be forced by setting the state `autoConnectFailed` to `true`.
* Klarna Checkout - requires store country to be 'SE', 'FI', 'NO', or 'NL'. 
* Klarna Payments - requires store country to be 'DK', 'DE', or 'AT'. 
* Square - requires store country to be 'US', 'CA', 'JP', 'GB', or 'AU' and having selected a "brick and mortar" option for the selling venues question in the store profiler.
* PayFast - requires store country to be South Africa.  

### Changelog Note:

Enhancement: Allow individual payment method setup in the onboarding task list
